diff --git a/arch/xtensa/Kconfig b/arch/xtensa/Kconfig
index 258bc91599..95d983daad 100644
--- a/arch/xtensa/Kconfig
+++ b/arch/xtensa/Kconfig
@@ -32,7 +32,7 @@ config XTENSA_USE_CORE_CRT1
 config XTENSA_ENABLE_BACKTRACE
 	bool "Backtrace on panic exception"
 	default y
-	depends on SOC_SERIES_ESP32 || SOC_FAMILY_INTEL_ADSP || SOC_XTENSA_DC233C || SOC_MIMXRT595S_F1
+	depends on SOC_FAMILY_ESPRESSIF_ESP32 || SOC_FAMILY_INTEL_ADSP || SOC_XTENSA_DC233C || SOC_MIMXRT595S_F1
 	help
 	  Enable this config option to print backtrace on panic exception
 
diff --git a/arch/xtensa/core/xtensa_backtrace.c b/arch/xtensa/core/xtensa_backtrace.c
index c96a3f7232..746c4230cb 100644
--- a/arch/xtensa/core/xtensa_backtrace.c
+++ b/arch/xtensa/core/xtensa_backtrace.c
@@ -8,7 +8,7 @@
 #include "xtensa_backtrace.h"
 #include <zephyr/sys/printk.h>
 #include <zephyr/arch/exception.h>
-#if defined(CONFIG_SOC_SERIES_ESP32)
+#if defined(CONFIG_SOC_FAMILY_ESPRESSIF_ESP32)
 #include <esp_memory_utils.h>
 #elif defined(CONFIG_SOC_FAMILY_INTEL_ADSP)
 #include "debug_helpers.h"
@@ -46,7 +46,7 @@ static inline bool xtensa_stack_ptr_is_sane(uint32_t sp)
 {
 	bool valid;
 
-#if defined(CONFIG_SOC_SERIES_ESP32)
+#if defined(CONFIG_SOC_FAMILY_ESPRESSIF_ESP32)
 	valid = esp_stack_ptr_is_sane(sp);
 #elif defined(CONFIG_SOC_FAMILY_INTEL_ADSP)
 	valid = intel_adsp_ptr_is_sane(sp);
@@ -67,7 +67,7 @@ static inline bool xtensa_stack_ptr_is_sane(uint32_t sp)
 
 static inline bool xtensa_ptr_executable(const void *p)
 {
-#if defined(CONFIG_SOC_SERIES_ESP32)
+#if defined(CONFIG_SOC_FAMILY_ESPRESSIF_ESP32)
 	return esp_ptr_executable(p);
 #elif defined(CONFIG_SOC_FAMILY_INTEL_ADSP)
 	return intel_adsp_ptr_executable(p);
